recipes-extended/xen/files/libxl-fixup-cmdline-ops.patch

```diff
@@ -0,0 +1,384 @@
+PATCHES
+################################################################################
+Index: xen-4.6.1/tools/libxl/libxl.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl.c
++++ xen-4.6.1/tools/libxl/libxl.c
+@@ -1160,6 +1160,24 @@ int libxl_domain_reboot(libxl_ctx *ctx,
+     return ret;
+ }
+ 
++int libxl_domain_sleep(libxl_ctx *ctx, uint32_t domid)
++{
++    GC_INIT(ctx);
++    int ret;
++    ret = libxl__domain_pvcontrol(gc, domid, "s3");
++    GC_FREE;
++    return ret;
++}
++
++int libxl_domain_hibernate(libxl_ctx *ctx, uint32_t domid)
++{
++    GC_INIT(ctx);
++    int ret;
++    ret = libxl__domain_pvcontrol(gc, domid, "hibernate");
++    GC_FREE;
++    return ret;
++}
++
+ static void domain_death_occurred(libxl__egc *egc,
+                                   libxl_evgen_domain_death **evg_upd,
+                                   const char *why) {
+Index: xen-4.6.1/tools/libxl/libxl.h
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl.h
++++ xen-4.6.1/tools/libxl/libxl.h
+@@ -1171,6 +1171,8 @@ int libxl_domain_remus_start(libxl_ctx *
+ 
+ int libxl_domain_shutdown(libxl_ctx *ctx, uint32_t domid);
+ int libxl_domain_reboot(libxl_ctx *ctx, uint32_t domid);
++int libxl_domain_sleep(libxl_ctx *ctx, uint32_t domid);
++int libxl_domain_hibernate(libxl_ctx *ctx, uint32_t domid);
+ int libxl_domain_destroy(libxl_ctx *ctx, uint32_t domid,
+                          const libxl_asyncop_how *ao_how)
+                          LIBXL_EXTERNAL_CALLERS_ONLY;
+Index: xen-4.6.1/tools/libxl/libxl_types.idl
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_types.idl
++++ xen-4.6.1/tools/libxl/libxl_types.idl
+@@ -154,6 +154,7 @@ libxl_trigger = Enumeration("trigger", [
+     (4, "INIT"),
+     (5, "RESET"),
+     (6, "S3RESUME"),
++    (7, "HIBERNATE"),
+     ])
+ 
+ libxl_tsc_mode = Enumeration("tsc_mode", [
+@@ -296,6 +297,7 @@ libxl_dominfo = Struct("dominfo",[
+     #
+     # Otherwise set to a value guaranteed not to clash with any valid
+     # LIBXL_SHUTDOWN_REASON_* constant.
++    ("power_command", uint32),
+     ("shutdown_reason", libxl_shutdown_reason),
+     ("outstanding_memkb",  MemKB),
+     ("current_memkb",   MemKB),
+Index: xen-4.6.1/tools/libxl/libxl_utils.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_utils.c
++++ xen-4.6.1/tools/libxl/libxl_utils.c
+@@ -93,6 +93,64 @@ int libxl_name_to_domid(libxl_ctx *ctx,
+     return ret;
+ }
+ 
++int libxl_get_acpi_state(libxl_ctx *ctx, int32_t domid, uint32_t *acpi_state)
++{
++    unsigned long hvm_s_state = 0;
++    int ret;
++    ret = xc_get_hvm_param(ctx->xch, domid, HVM_PARAM_ACPI_S_STATE, &hvm_s_state);
++    if (ret < 0) { 
++        hvm_s_state = 9; //since unsigned, use invalid acpi state
```

Comment:
Would it not be better to use a define instead of an explicit assign of an arbitrary value? That way it doesn't matter what the value is and it makes the code easier to reason.